tests/spread/plugins/go/rpath-rewrite-cgo/task.yaml

```diff
@@ -0,0 +1,36 @@
+summary: Build and run Go snaps with dynamic executables
+
+environment:
+  SNAP_DIR: ../snaps/go-cgo
+
+prepare: |
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+  # Set the confinement type of this snap to classic
+  sed -i 's/\(confinement.*\)strict/\1classic/;/^confinement:/a base: core18' "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+  cd "$SNAP_DIR"
+
+  snapcraft prime
+
+  if ! snap list go; then
+    echo "The go snap from the Snap Store should have been installed."
+    exit 1
+  fi
+
+  # Ensure binaries are properly patched
+  patchelf --print-rpath prime/bin/go-cgo | MATCH "/snap/core[0-9]*/current"
```

Comment:
make this `/snap/snapcraft/current/bin/patchelf`

Reply:
and if you really wanted some independence here, you you install binutils and use readelf for this :-)